src/d3a/d3a_core/simulation.py

```diff
@@ -241,24 +222,18 @@ def deactivate_areas(self, area):
         for child in area.children:
             self.deactivate_areas(child)
 
-    def run(self, resume=False) -> (Period, duration):
+    def run(self, initial_slot=0):
         self.sim_status = "running"
-        if resume:
-            log.critical("Resuming simulation")
-            self._info()
         self.is_stopped = False
         while True:
-            if resume:
-                # FIXME: Fix resume time calculation
-                if self.run_start is None or self.paused_time is None:
-                    raise RuntimeError("Can't resume without saved state")
-                slot_resume, tick_resume = divmod(self.area.current_tick,
-                                                  self.simulation_config.ticks_per_slot)
-            else:
+            if initial_slot == 0:
                 self.run_start = DateTime.now(tz=TIME_ZONE)
                 self.paused_time = 0
-                slot_resume = tick_resume = 0
+                slot_resume = 0
+            else:
+                slot_resume = initial_slot
```

Comment:
I think `slot_resume ` could be pulled out of this if else block, simply via always setting `slot_resume = initial_slot`

Reply:
Agreed, fixing this